config/config.go

```diff
@@ -819,6 +822,53 @@ func (sc *SessionsConfig) Validate(path string) error {
 	return nil
 }
 
+// TrafficTunnelEndpoint is an endpoint for tunneling traffic.
+type TrafficTunnelEndpoint struct {
+	// Port is the port which can be tunneled to/from.
+	Port int
+	// ConnectionTimeout is the timeout with which we will attempt to connect to the port.
+	// If set to 0 or not specified, a default connection timeout of 10 seconds will be used.
```

Comment:
We were using a hard-coded 10s timeout before. I could see an argument for `0` or unspecified meaning infinite timeout, but 10s seems like a sensible default IMO. No strong opinion here, though.